### PR TITLE
Add object actions - create folder and delete object(s)

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
https://issues.redhat.com/browse/RHSTOR-6511
https://issues.redhat.com/browse/RHSTOR-6508

Screenshot 2024-10-03 at 6 55 43 PM
Screenshot 2024-10-03 at 6 55 57 PM
Screenshot 2024-10-03 at 6 56 19 PM
Screenshot 2024-10-07 at 7 41 41 PM
Screenshot 2024-10-08 at 10 34 31 AM
Screenshot 2024-10-08 at 10 35 37 AM
Screenshot 2024-10-08 at 12 50 31 PM
Screenshot 2024-10-08 at 12 50 48 PM

### CODE REVIEW

File: locales/en/plugin__odf-console.json [hunk 1]
```diff
@@ -1128,6 +1128,12 @@
   "Delete objects": "Delete objects",
   "Actions": "Actions",
   "Create folder": "Create folder",
+  "Failed to delete {{ count }} objects from the bucket. View deletion summary for details._one": "Failed to delete {{ count }} objects from the bucket. View deletion summary for details.",
```
Comment (alfonsomthd) [inline locales/en/plugin__odf-console.json:1131]: For `._one` shouldn't be the translation in singular?
```suggestion
  "Failed to delete {{ count }} object from the bucket. View deletion summary for details._one": "Failed to delete {{ count }} objects from the bucket. View deletion summary for details.",
```

File: packages/odf/modals/s3-browser/delete-objects/DeleteObjectsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+      } catch (err) {
+        setInProgress(false);
+        setError(err);
+      }
+    };
+
+    return (
+      <Modal
+        title={t('Delete object?')}
+        titleIconVariant="warning"
```
Comment (alfonsomthd) [inline packages/odf/modals/s3-browser/delete-objects/DeleteObjectsModal.tsx:159]: Do we any variant enum for this?

File: packages/shared/src/s3/commands.ts [hunk 3]
```diff
@@ -51,4 +53,7 @@ export class S3Commands {
 
   getObject: GetObject = (input) =>
     this.s3Client.send(new GetObjectCommand(input));
+
+  deleteObjects: DeleteObjects = (input) =>
```
Comment (alfonsomthd) [inline packages/shared/src/s3/commands.ts:57]: Just an optional comment: better if we place the methods alphabetically: easier to find.

File: packages/shared/src/s3/types.ts [hunk 4]
```diff
@@ -37,3 +39,7 @@ export type GetSignedUrl = (
 export type GetObject = (
   input: GetObjectCommandInput
 ) => Promise<GetObjectCommandOutput>;
+
+export type DeleteObjects = (
```
Comment (alfonsomthd) [inline packages/shared/src/s3/types.ts:43]: Just an optional comment: better if we place them alphabetically: easier to find.

File: packages/odf/modals/s3-browser/delete-objects/DeleteObjectsModal.tsx (hunk 2 above)
Comment (SanjalKatiyar) [reply packages/odf/modals/s3-browser/delete-objects/DeleteObjectsModal.tsx:159]: Didn't find any :/

File: packages/shared/src/list-page/ListFilter.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { getName } from '@odf/shared/selectors';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import { SearchInput, SearchInputProps } from '@patternfly/react-core';
+
+type TListFilter = {
```
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/shared/src/list-page/ListFilter.tsx:8]: make sense...

File: packages/shared/src/s3/commands.ts (hunk 3 above)
Comment (SanjalKatiyar) [reply packages/shared/src/s3/commands.ts:57]: make sense, will do in a separate PR...

File: packages/shared/src/s3/types.ts (hunk 4 above)
Comment (SanjalKatiyar) [reply packages/shared/src/s3/types.ts:43]: will do in a separate PR...

**Pull request merged**